OpenDreamServer/Dream/Objects/DreamList.cs

```diff
@@ -119,10 +118,10 @@ class DreamList : DreamObject {
         }
 
         public static bool IsValidAssociativeKey(DreamValue key) {
-            return key.Value != null && key.IsType( DreamValue.DreamValueType.String |
-                                                    DreamValue.DreamValueType.DreamPath |
-                                                    DreamValue.DreamValueType.DreamObject |
-                                                    DreamValue.DreamValueType.DreamResource);
+            return key != null && (key == DreamValue.Null || key.IsType(DreamValue.DreamValueType.String |
```

Comment:
DreamValue is a struct, which cannot be null.
```suggestion
            return (key == DreamValue.Null || key.IsType(DreamValue.DreamValueType.String |
```